app/lib/stats/views/usage.rb

```diff
@@ -158,6 +136,26 @@ def extract_range_and_granularity(options)
 
       protected
 
+      def application_result
+        return {} unless @cinstance
+
+        plan = @cinstance.plan
+        account = @cinstance.user_account
+
+        result_application = {
+          id:          @cinstance.id,
+          name:        @cinstance.name,
+          state:       @cinstance.state,
+          description: @cinstance.description,
+          service: { id: @cinstance.service_id }
+        }
+
+        result_application[:plan]    = {id: plan.id,    name: plan.name}    if plan
+        result_application[:account] = {id: account.id, name: account.name} if account
```

Comment:
Couldn't this be masking a bigger problem? Should we really have applications whose plan was deleted or missing a buyer account?

Besides, could the omitting of these attributes when the rest of info of the application is available cause side-effects in the UI or API?